andela/travel_tool_front
# #161595854 User can add a travel document

#### What does this PR do?
This PR implements the add/upload document feature.

#### Description of Task to be completed?
- A user can upload the document to the cloud.
- A user can add the document to the database

#### How should this be manually tested?
- A Travel Admin/Super Admin for a location
 - Clone the repo - `git clone https://github.com/andela/travel_tool_front.git`
      - Change into the project directory
      - Run the server `make start`
      - Login and navigate the documents page
      - Add a new document using the add file button
      - A modal will appear where you click the upload file button to choose the file.
      - In the modal type in the name of the document in the input field
      - Click the add button in the modal to upload/add the document
      - Upload image file or PDF.
      - A toast message should be displayed to indicate success or failure in case of any errors. 

#### Any background context you want to provide?
- You need to clone the [backend](https://github.com/andela/travel_tool_back.git) and switch to the ft-add-travel-document-161595854 branch to test this functionality.
- Then run ```yarn db:rollmigrate```
- If using yarn start use the following .env variables
```
REACT_APP_CLOUNDINARY_API=https://api.cloudinary.com/v1_1/authors-haven/image/upload
REACT_APP_PRESET_NAME=iwbpjk8d
```

#### What are the relevant pivotal tracker stories?
[#161595854](https://www.pivotaltracker.com/story/show/161595854)

#### Screenshots (if appropriate)
<img width="1440" alt="screen shot 2018-11-07 at 15 41 14" src="https://user-images.githubusercontent.com/12511966/48132169-ab858100-e2a3-11e8-8720-e5fd9f6709fa.png">
<img width="1440" alt="screen shot 2018-11-07 at 15 41 29" src="https://user-images.githubusercontent.com/12511966/48132184-b3452580-e2a3-11e8-9342-78fc66aa00a7.png">


#### Questions:
